types/part_set.go

```diff
@@ -309,12 +322,21 @@ func (ps *PartSet) AddPart(part *Part) (bool, error) {
 		return false, fmt.Errorf("%w: part.Proof.Verify(ps.Hash(), part.Bytes) = %v", ErrPartSetInvalidProof, err)
 	}
 
-	return ps.AddPartWithoutProof(part)
+	// Add part
+	ps.parts[part.Index] = part
+	ps.partsBitArray.SetIndex(int(part.Index), true)
+	ps.count++
+	ps.byteSize += int64(len(part.Bytes))
+	return true, nil
 }
 
+// AddPartWithoutProof is a Celestia-specific method that was inspired by the
+// AddPart method. AddPartWithoutProof ignores the Proof inside part.
 func (ps *PartSet) AddPartWithoutProof(part *Part) (bool, error) {
-	if part == nil {
-		return false, errors.New("nil part")
+	// TODO: remove this? would be preferable if this only returned (false, nil)
+	// when its a duplicate block part
+	if ps == nil {
```

Comment:
is this being hit? we shouldn't ever I don't think. The only reason it returns an error afaiu is to avoid adding a panic (and thus attack surface).

if it is not being hit then I don't think that we should change this.

Reply:
Historical context:
- https://github.com/celestiaorg/celestia-core/pull/1513
- https://github.com/celestiaorg/celestia-core/pull/1513/files#r1848349010

TBH it's confusing and I think it's clearer to have an exact copy of `AddPart` from upstream and an `AddPartWithoutProof` that is Celestia-specific that skips the proving